### PR TITLE
CEDS-1484 Amend a rejected declaration

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/controllers/SubmissionsController.scala [hunk 1]
```diff
@@ -45,4 +49,15 @@ class SubmissionsController @Inject()(
     } yield Ok(submissionsPage(result.toSeq))
   }
 
+  def amend(id: String): Action[AnyContent] = authenticate.async { implicit request =>
+    customsDeclareExportsConnector.findDeclaration(authenticatedSessionId, id) flatMap {
```
Comment (bambuchaAdm) [inline app/controllers/SubmissionsController.scala]: there is `sessionId` on request - we don't have to use `SessionIdAware` trait

File: app/controllers/SubmissionsController.scala [hunk 2]
```diff
@@ -45,4 +49,15 @@ class SubmissionsController @Inject()(
...
+      case Some(declaration) =>
+        exportsCacheService
+          .update(authenticatedSessionId, declaration)
```
Comment (bambuchaAdm) [inline app/controllers/SubmissionsController.scala]: is here possible `sessionId` and `declaration.sessionId` could be different ?

File: app/services/SubmissionService.scala [hunk 3]
```diff
@@ -52,7 +52,7 @@ class SubmissionService @Inject()(
     val timerContext = exportsMetrics.startTimer(submissionMetric)
     auditService.auditAllPagesUserInput(getCachedData(exportsDeclaration))
     for {
-      _ <- exportsConnector.create(exportsDeclaration).recover {
+      _ <- exportsConnector.createDeclaration(exportsDeclaration).recover {
```
Comment (bambuchaAdm) [inline app/services/SubmissionService.scala]: this is out of scope of this PR but here we would remove cache even connector would return an error

File: app/views/declaration/summary/summary_page.scala.html [hunk 4]
```diff
@@ -74,14 +80,20 @@
 
     @items_section(suppDecData.items)
 
-    @helper.form(action = SummaryController.submitSupplementaryDeclaration(), 'id -> "summary") {
-        @helper.CSRF.formField
+    @if(mode == Mode.AmendMode) {
```
Comment (bambuchaAdm) [inline app/views/declaration/summary/summary_page.scala.html]: @ed0906 what you think about 
```
mode.onAmend {
  <a class="button mt-3" href="@routes.DispatchLocationController.displayPage()">@messages("site.continue")</a>
}
```

File: test/unit/controllers/declaration/SealControllerSpec.scala [hunk 5]
```diff
@@ -140,6 +139,7 @@ class SealControllerSpec extends ControllerSpec with ScalaFutures with ErrorHand
         val result = controller.submitForm()(postRequestAsFormUrlEncoded(body: _*))
 
         status(result) must be(SEE_OTHER)
+        redirectLocation(result) must be(Some(controllers.declaration.routes.SummaryController.displayPage(Mode.NormalMode).url))
```
Comment (PatrykRudnicki) [inline test/unit/controllers/declaration/SealControllerSpec.scala]: 👍

File: app/views/submission_notifications.scala.html [hunk 6]
```diff
@@ -17,33 +17,42 @@
...
-            }
-        </table>
-    </div>
+        </thead>
+
+        <tbody>
+        @for(notification <- notifications){
+            <tr class="table-row">
+                <td class="table-cell">@notification.status.toString</td>
+                <td class="table-cell">@DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm").format(notification.dateTimeIssued)</td>
```
Comment (bambuchaAdm) [inline app/views/submission_notifications.scala.html]: Please do not build formatter at each request.

File: conf/app.routes [hunk 7]
```diff
@@ -10,12 +10,14 @@ GET        /unauthorised                                controllers.Unauthorised
 
 GET        /notifications                               controllers.NotificationsController.listOfNotifications()
 
-GET        /notifications/:mrn                          controllers.NotificationsController.listOfNotificationsForSubmission(mrn: String)
+GET        /notifications/:id                          controllers.NotificationsController.listOfNotificationsForSubmission(id: String)
```
Comment (bambuchaAdm) [inline conf/app.routes]: On backend we moved to `submissions/:id/notifications` maybe on frontend we should also

File: app/controllers/SubmissionsController.scala (hunk 1 above)
Comment (ed0906) [reply app/controllers/SubmissionsController.scala]: We probably dont need this trait any more at all

File: app/controllers/SubmissionsController.scala (hunk 2 above)
Comment (ed0906) [reply app/controllers/SubmissionsController.scala]: No, session id isn't stored in the back end so the only way it could be populated is from the request

Comment (ed0906) [reply app/controllers/SubmissionsController.scala]: I'm not sure we actually need `declaration.sessionId` any more to be honest

File: app/services/SubmissionService.scala (hunk 3 above)
Comment (ed0906) [reply app/services/SubmissionService.scala]: I think the "Save and Resume" story will mean we no longer need the cache. I discussed with Steve (who is planning to pick this work up) that we can potentially remove the cache as part of it

File: app/views/declaration/summary/summary_page.scala.html (hunk 4 above)
Comment (ed0906) [reply app/views/declaration/summary/summary_page.scala.html]: Hmmmm, I don't think its what we would intuitively think to use

File: app/views/submission_notifications.scala.html (hunk 6 above)
Comment (ed0906) [reply app/views/submission_notifications.scala.html]: Sorted

File: conf/app.routes (hunk 7 above)
Comment (ed0906) [reply conf/app.routes]: Good suggestion

Comment (ed0906) [reply conf/app.routes]: Done

File: app/views/declaration/summary/summary_page.scala.html (hunk 4 above)
Comment (bambuchaAdm) [reply app/views/declaration/summary/summary_page.scala.html]: It same concept like foreach on `Option` but for enumeration

File: app/controllers/SubmissionsController.scala (hunk 2 above)
Comment (bambuchaAdm) [reply app/controllers/SubmissionsController.scala]: sessionId was and old `Id` for declaration draft

**Pull request merged**